drivers/bus/acpi/main.c

```diff
@@ -247,10 +239,41 @@ ACPIDispatchDeviceControl(
     {
        switch (irpStack->Parameters.DeviceIoControl.IoControlCode)
        {
+            case IOCTL_ACPI_ASYNC_EVAL_METHOD:
+            {
+                ASSERT(KeGetCurrentIrql() <= DISPATCH_LEVEL);
+
+                if (KeGetCurrentIrql() == DISPATCH_LEVEL)
+                {
+                    PEVAL_WORKITEM_DATA workItemData;
+
+                    workItemData = ExAllocatePoolWithTag(NonPagedPool,
+                                                         sizeof(*workItemData),
+                                                         'ipcA');
+                    if (!workItemData)
+                    {
+                        status = STATUS_INSUFFICIENT_RESOURCES;
+                        break;
+                    }
+                    workItemData->Irp = Irp;
+                    workItemData->DeviceData = (PPDO_DEVICE_DATA)commonData;
+
+                    ExInitializeWorkItem(&workItemData->WorkQueueItem,
+                                         Bus_PDO_EvalMethodWorker,
+                                         workItemData);
+                    ExQueueWorkItem(&workItemData->WorkQueueItem, DelayedWorkQueue);
+
+                    status = STATUS_PENDING;
+                    break;
+                }
+                __fallthrough;
+            }
            case IOCTL_ACPI_EVAL_METHOD:
+            {
```

Comment:
Just to confirm: here you put the brackets with correct indentation?

Reply:
Yes